cranelift/codegen/src/souper_harvest.rs

```diff
@@ -0,0 +1,525 @@
+                    }
+                    (ir::Opcode::Select, _) => {
+                        let a = arg(allocs, 0);
+                        let b = arg(allocs, 1);
+                        let c = arg(allocs, 2);
+                        ast::Instruction::Select { a, b, c }.into()
+                    }
+                    (ir::Opcode::Uextend, _) => {
+                        let a = arg(allocs, 0);
+                        ast::Instruction::Zext { a }.into()
+                    }
+                    (ir::Opcode::Sextend, _) => {
+                        let a = arg(allocs, 0);
+                        ast::Instruction::Sext { a }.into()
+                    }
+                    (ir::Opcode::Trunc, _) => {
+                        let a = arg(allocs, 0);
+                        ast::Instruction::Trunc { a }.into()
+                    }
+                    (ir::Opcode::Icmp, ir::InstructionData::IntCompare { cond, .. })
+                    | (ir::Opcode::IcmpImm, ir::InstructionData::IntCompare { cond, .. }) => {
+                        let a = arg(allocs, 0);
+                        let b = arg(allocs, 1);
+                        match cond {
+                            ir::condcodes::IntCC::Equal => ast::Instruction::Eq { a, b }.into(),
+                            ir::condcodes::IntCC::NotEqual => ast::Instruction::Ne { a, b }.into(),
+                            ir::condcodes::IntCC::UnsignedLessThan => {
+                                ast::Instruction::Ult { a, b }.into()
+                            }
+                            ir::condcodes::IntCC::SignedLessThan => {
+                                ast::Instruction::Slt { a, b }.into()
+                            }
+                            ir::condcodes::IntCC::UnsignedLessThanOrEqual => {
+                                ast::Instruction::Sle { a, b }.into()
+                            }
+                            ir::condcodes::IntCC::SignedLessThanOrEqual => {
+                                ast::Instruction::Sle { a, b }.into()
+                            }
+                            _ => ast::AssignmentRhs::Var,
+                        }
+                    }
+                    (ir::Opcode::Popcnt, _) => {
+                        let a = arg(allocs, 0);
+                        ast::Instruction::Ctpop { a }.into()
+                    }
+                    (ir::Opcode::Bitrev, _) => {
+                        let a = arg(allocs, 0);
+                        ast::Instruction::BitReverse { a }.into()
+                    }
+                    (ir::Opcode::Clz, _) => {
+                        let a = arg(allocs, 0);
+                        ast::Instruction::Ctlz { a }.into()
+                    }
+                    (ir::Opcode::Ctz, _) => {
+                        let a = arg(allocs, 0);
+                        ast::Instruction::Cttz { a }.into()
+                    }
+                    // TODO: ir::Opcode::IaddCarry
```

Comment:
If, indeed, there should be an `IaddCout` above, then this should be updated too